graylog2-web-interface/src/components/enterprise/EnterpriseFreeLicenseForm.jsx

```diff
@@ -0,0 +1,127 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+
+import { Button, ButtonToolbar, Col, Row } from 'components/graylog';
+import { Input } from 'components/bootstrap';
+import lodash from 'lodash';
+
+const FORM_FIELDS = ['firstName', 'lastName', 'email', 'company'];
+
+export default class EnterpriseFreeLicenseForm extends React.Component {
+  static propTypes = {
+    onSubmit: PropTypes.func.isRequired,
+  };
+
+  constructor(props) {
+    super(props);
+    this.state = {
+      formFields: {
+        firstName: '',
+        lastName: '',
+        email: '',
+        company: '',
+      },
+      isSubmitting: false,
+    };
+  }
+
+  clearValues = (callback) => {
+    const clearedFields = FORM_FIELDS.reduce((acc, key) => Object.assign(acc, { [key]: '' }), {});
+    this.setState({ formFields: clearedFields }, callback);
+  };
+
+  handleInput = (key) => {
+    return (event) => {
+      const { formFields } = this.state;
+      const newFormFields = Object.assign(formFields, { [key]: event.target.value });
+      this.setState({ formFields: newFormFields });
+    };
+  };
+
+  formIsInvalid = () => {
+    const { isSubmitting, formFields } = this.state;
+
+    return isSubmitting || !lodash.isEmpty(FORM_FIELDS.filter((key) => lodash.isEmpty(lodash.trim(formFields[key]))));
+  };
+
+  submitForm = (event) => {
+    event.preventDefault();
+
+    const { onSubmit } = this.props;
+    const { formFields } = this.state;
+
+    // First set "submitting" status to make sure we disable the submit button (avoid double-click)
+    this.setState({ isSubmitting: true }, () => {
+      onSubmit(formFields, (success) => {
+        if (success) {
+          // Clear form before unsetting "submitting" status, again, to avoid double-click
+          this.clearValues(() => {
+            this.setState({ isSubmitting: false });
+          });
+        } else {
+          this.setState({ isSubmitting: false });
+        }
+      });
+    });
+  };
+
+  resetForm = () => {
+    this.clearValues();
+  };
+
+  render() {
+    const { formFields: { firstName, lastName, company, email } } = this.state;
+
+    return (
+      <form onSubmit={this.submitForm}>
+        <Row>
+          <Col md={5}>
```

Comment:
This leads to a truncated form:
![image](https://user-images.githubusercontent.com/52014/80593049-c3fdd400-8a20-11ea-8779-67402878ae4e.png)

Can we make it a bit wider?

Reply:
Good catch. Yes! :+1: